src/importer/ImportSession.ts

```diff
@@ -107,7 +110,11 @@ export class ImportSession extends TypedEventManager<IImportSessionEvents> {
 
   private async subscribeToBatchStatus(): Promise<void> {
     return await this.api.subscribeBatchStatusUpdated(this.batchId, async (batch) => {
-      if (batch.status === 'evaluate') {
+      // only emit this emit once
+      // otherwise, many RecordChunkIterators could be created
+      // in this case, the record chunks passed to onData could be unreliable
+      if (batch.status === 'evaluate' && this.meta.status !== 'evaluate') {
```

Comment:
emitting multiple `evaluate` events is problematic; especially if we run into a long running operation or if WSS fails and we need to poll. If this is emitted a second time, it will create a new iterator which will make customer's onData events fire multiple times, potentially for the same chunk

Reply:
Hmmm, interesting. But you can actually resubmit many times. That's the whole point of this event.

Reply:
Hmmm you are correct, we need this for resubmitting but in situations where we are polling this is causing a lot of issues creating many iterators at the wrong time